File: impact_lab/src/types/patient.ts
export type CESFAMSector = 'SECTOR_VERDE' | 'SECTOR_AZUL' | 'SECTOR_ROJO' | 'SECTOR_AMARILLO';

export type ContraloriaStatus = 
  | 'PENDIENTE'
  | 'APROBADO'
  | 'REQUIERE_REVISION'
  | 'OBSERVADO';

export type RiskLevel = 'CRITICO' | 'ALTO' | 'MEDIO' | 'BAJO';

export interface DecompensationFactor {
  code: string;
  label: string;
  severity: 'ALTA' | 'MEDIA' | 'LEVE';
  value: string;
}

export interface NT118RiskScore {
  totalScore: number;
  riskLevel: RiskLevel;
  hba1cScore: number;
  renalsScore: number; // VFG & Proteinuria
  cvScore: number;    // Hipertensión & eventos previos
  socialScore: number; // Determinantes sociales y ruralidad
  decompensations: DecompensationFactor[];
}

export interface AuditLogEntry {
  id: string;
  timestamp: string;
  userName: string;
  userRole: string;
  action: string;
  previousStatus: ContraloriaStatus;
  newStatus: ContraloriaStatus;
  clinicalNote: string;
}

export interface PriorizacionRequest {
  patient_id?: string;
  patientId?: string;
  hba1c: number;
  systolic_bp?: number;
  systolicBP?: number;
  diastolic_bp?: number;
  diastolicBP?: number;
  vfg: number;
  has_foot_ulcer?: boolean;
  hasFootUlcer?: boolean;
  has_retinopathy?: boolean;
  hasRetinopathy?: boolean;
  days_in_waiting_list?: number;
  daysInWaitingList?: number;
  age?: number;
  gender?: string;
  sector?: string;
  cesfam_name?: string;
  cesfamName?: string;
}

export interface Patient {
  id: string;
  rut: string;
  fullName: string;
  age: number;
  gender: 'M' | 'F';
  sector: CESFAMSector;
  cesfamName: string;
  
  // Parámetros clínicos APS
  hba1c: number;             // % Hemoglobina Glicosilada
  systolicBP: number;        // Presión Sistólica
  diastolicBP: number;       // Presión Diastólica
  vfg: number;               // Velocidad Filtración Glomerular (mL/min/1.73m2)
  hasFootUlcer: boolean;     // Pie diabético activo
  hasRetinopathy: boolean;   // Retinopatía diabética
  
  // Tiempos y priorización algorítmica
  daysInWaitingList: number;
  nt118Risk: NT118RiskScore;
  priorityPosition: number;
  previousPriorityPosition: number; // Trazabilidad de cambio
  
  // Estado de Contraloría Médica
  contraloriaStatus: ContraloriaStatus;
  assignedPhysician?: string;
  lastReviewDate?: string;
  auditHistory: AuditLogEntry[];
}

File: impact_lab/backend/app/schemas/patient.py
from pydantic import BaseModel, Field, ConfigDict
from typing import List, Optional, Literal
from app.schemas.priorizacion import PriorizacionResponse, CamelModel

class AuditLogEntry(CamelModel):
    id: str
    timestamp: str
    user_name: str
    user_role: str
    action: str
    previous_status: str
    new_status: str
    clinical_note: str

class ContraloriaUpdatePayload(CamelModel):
    new_status: str  # "PENDIENTE" | "APROBADO" | "RECHAZADO" | "DERIVADO" | "REQUIERE_REVISION" | "OBSERVADO"
    clinical_note: str
    physician_name: str
    physician_role: str

class Patient(CamelModel):
    id: str
    rut: str
    rut_hash: Optional[str] = None
    full_name: str
    age: int
    gender: str
    sector: str
    cesfam_name: str
    hba1c: float
    systolic_bp: int
    diastolic_bp: int
    vfg: float
    has_foot_ulcer: bool
    has_retinopathy: bool
    days_in_waiting_list: int
    nt118_risk: PriorizacionResponse
    priority_position: int
    previous_priority_position: int
    contraloria_status: str
    assigned_physician: Optional[str] = None
    last_review_date: Optional[str] = None
    audit_history: List[AuditLogEntry] = Field(default_factory=list)
